### DISCUSSION TITLE
When running multiple queries in a script the "References" tab does not work

### DISCUSSION THREAD

**Discussion by Amitkad** [dbeaver/dbeaver#36389] [Q&A] [answered]
### Description

Steps:
1. Open a new "script" window
2. Write two queries
3. Select all and press Ctrl+Enter to run it

Expected:

* Each query result will be shown in a different window
* For each query user would be able to navigate to references using the "References" tab

Actual:

* References tab dropdown list is empty

Additional details:
* When running each query separately the references tab for each works
* query exmaple I used:

select * from time_slot ts WHERE DATE(ts.date) = current_date  + interval '1 days';
select * from route r WHERE DATE(r.date) = current_date  + interval '1 days';
 Result:
![image](https://github.com/user-attachments/assets/0b2c5fd1-2e14-461a-be9d-1d2ef86b0c0e)

### DBeaver Version

Community Edition 24.2.0.202409011551

### Operating System

Windows 10 home

### Database and driver

psql (PostgreSQL) 13.2

**Comment by ShadelessFox** [comment, marked as answer]
Hello @Amitkad,

When you select a block of text containing multiple queries and hit **Ctrl+Enter**, DBeaver submits that entire blob to the driver, which in turn sends it to the database. Depending on a particular database/driver, it might be handled differently. The most problematic part is the metadata of the result set. Under some circumstances, DBeaver cannot correlate that metadata with the actual structure of the database, which is required for providing features like data editing, reference navigation, etc.

To work around this, you can execute each query separately using **Ctrl+\\** or execute the entire script using **Alt+X** - DBeaver will submit each query separately.

TL;DR: Use **Alt+X** to execute multiple queries.

**Reply by ask9** [reply]
I think you meant Ctrl + \ 
Not Ctrl + /

**Reply by ShadelessFox** [reply]
Right, thanks for noticing.

**Reply by Amitkad** [reply]
Thanks a lot for the answer. indeed alt+x solved my issue.

In terms of improving the product experience for all users (not only the ones that read in this forum), I think it is still something that should be resolved (we can call it a UX bug, and not technical one...).
Well, I don't know the disadvantages of it, but for example make alt+x as the default (like what ctrl+enter does today), or at least have a different description on the key shortcut.
